src/main/java/swinglib/JPanelBuilder.java

```diff
@@ -0,0 +1,191 @@
+package swinglib;
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.GridBagLayout;
+import java.awt.GridLayout;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.swing.BoxLayout;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.border.EmptyBorder;
+
+import org.apache.commons.math3.util.Pair;
+
+/**
+ * Example usage:
+ * <code><pre>
+ *   final JPanel panel = JPanelBuilder.builder()
+ *       .gridLayout(2, 1)
+ *       .add(new JLabel("")
+ *       .add(new JLabel("")
+ *       .build();
+ * </pre></code>
+ */
+public class JPanelBuilder {
+
+  private final Collection<Pair<Component, BorderLayoutPosition>> components = new ArrayList<>();
+  private Layout layout = Layout.DEFAULT;
+  private int gridRows;
+  private int gridColumns;
+  private BorderType borderType;
+  private int borderWidth;
+  private Float xAlignment;
+
+  private JPanelBuilder() {}
+
+  public static JPanelBuilder builder() {
+    return new JPanelBuilder();
+  }
+
+  /**
+   * Constructs a Swing JPanel using current builder values.
+   * Values that must be set: (requires no values to be set)
+   */
+  public JPanel build() {
+    final JPanel panel = new JPanel();
+    if (borderType != null) {
+      switch (borderType) {
+        case EMPTY:
+        default:
+          panel.setBorder(new EmptyBorder(borderWidth, borderWidth, borderWidth, borderWidth));
+          break;
+      }
+    }
+    switch (layout) {
+      case GRID:
+        panel.setLayout(new GridLayout(gridRows, gridColumns));
+        break;
+      case GRID_BAG:
+        panel.setLayout(new GridBagLayout());
+        break;
+      case BOX_LAYOUT_HORIZONTAL:
+        panel.setLayout(new BoxLayout(panel, BoxLayout.X_AXIS));
+        break;
+      case BOX_LAYOUT_VERTICAL:
+        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
+        break;
+      case DEFAULT:
+        panel.setLayout(new BorderLayout());
+        break;
+    }
+    for (final Pair<Component, BorderLayoutPosition> child : components) {
+      if (child.getSecond() == BorderLayoutPosition.DEFAULT) {
+        panel.add(child.getFirst());
+      } else {
+        switch (child.getSecond()) {
+          case CENTER:
+            panel.add(child.getFirst(), BorderLayout.CENTER);
+            break;
+          case SOUTH:
+            panel.add(child.getFirst(), BorderLayout.SOUTH);
+            break;
+          case NORTH:
+            panel.add(child.getFirst(), BorderLayout.NORTH);
+            break;
+          case WEST:
+            panel.add(child.getFirst(), BorderLayout.WEST);
+            break;
+          case EAST:
+            panel.add(child.getFirst(), BorderLayout.EAST);
+            break;
+          default:
+            panel.add(child.getFirst());
+            break;
+        }
+      }
+    }
+    if(xAlignment != null) {
```

Comment:
Checkstyle: "WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)"